src/test/java/com/coveo/spillway/SpillwayTest.java

```diff
@@ -118,20 +121,17 @@ public void simpleLimit() {
     assertThat(spillway.tryCall(john)).isFalse(); // Third tryCall fails
   }
 
-  @Test
-  public void multipleLimitsWithOverlap() {
+  @Test(expected = SpillwayLimitWithSameNameException.class)
+  public void multipleLimitsWithOverlap() throws Exception {
```

Comment:
I don't see the point of using the same name. The situation before was just not predictable since it would chose the limit to apply a bit arbitrarily.
In this case just use two limits since we now check the limits before incrementing them.